src/scenic/core/object_types.py

```diff
@@ -3,666 +3,1760 @@
+          value is the global coordinate system, where an object is flat in the XY plane,
+          facing North.
+        orientation (`Orientation`; dynamic; final): The orientation of the `OrientedPoint` relative
+          to the global coordinate system. Derived from the :prop:`yaw`, :prop:`pitch`, 
+          :prop:`roll`, and :prop:`parentOrientation` of this `OrientedPoint` and non-overridable.
+        heading (float; dynamic; final): Yaw value of this `OrientedPoint` in the global coordinate
+          system. Derived from :prop:`orientation` and non-overridable.
+        viewAngles (tuple[float,float]): Horizontal and vertical view angles of this `OrientedPoint`
+          in radians. Horizontal view angle can be up to 2π and vertical view angle can be
+          up to π. Values greater than these will be truncated. Default value is (2π, π)
+        orientationStdDev (float): Standard deviation of Gaussian noise to add to this
+          object's Euler angles (yaw, pitch, roll) when mutation is enabled with scale 1.
+          Default value (5π, 0, 0), mutating only the :prop:`yaw` of this `OrientedPoint`.
+    """
+    _scenic_properties = {
+        'yaw': PropertyDefault((), {'dynamic'}, lambda self: 0),
+        'pitch': PropertyDefault((), {'dynamic'}, lambda self: 0),
+        'roll': PropertyDefault((), {'dynamic'}, lambda self: 0),
+        'parentOrientation': Orientation.fromEuler(0, 0, 0),
+
+        'orientation': PropertyDefault(
+            {'yaw', 'pitch', 'roll', 'parentOrientation'},
+            {'dynamic', 'final'},
+            lambda self: (Orientation.fromEuler(self.yaw, self.pitch, self.roll)
+                      * self.parentOrientation)
+        ),
+        'heading': PropertyDefault({'orientation'}, {'dynamic', 'final'},
+            lambda self: self.yaw if alwaysGlobalOrientation(self.parentOrientation) else self.orientation.yaw),
+
+        'viewAngle': math.tau, # Primarily for backwards compatibility. Set viewAngles instead.
+        'viewAngles': PropertyDefault(('viewAngle',), set(), lambda self: (self.viewAngle, math.pi)),
+
+        'mutator': PropertyDefault({'orientationStdDev'}, {'additive'},
+            lambda self: OrientationMutator(self.orientationStdDev)),
+        'orientationStdDev': (math.radians(5),0,0),
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if self.viewAngles[0] > math.tau or self.viewAngles[1] > math.pi:
+            warnings.warn("ViewAngles can not have values greater than (math.tau, math.pi). Truncating values...")
+            self.viewAngles = (min(self.viewAngles[0], math.tau), min(self.viewAngles[1], math.pi))
+
+    @cached_property
+    def visibleRegion(self):
+        """The :term:`visible region` of this object.
+
+        The visible region of an `OrientedPoint`, whose shape depends on
+        :prop:`viewAngles`. For more details, see `DefaultViewRegion`.
```

Comment:
It would be good to include a brief description of the shape of the visible region here, since this text will appear in the "Objects and Classes" page of the documentation and the gory details (which you get at `DefaultViewRegion`) are too much for that. I think you can say that in general it's a capped rectangular pyramid subtending an angle of  `viewAngles[0]` horizontally and `viewAngles[1]` vertically, as long as those angles are less than 180 degrees, but that various kinds of wrap-around regions are also possible.